Protocol changes for 1.21.80

dries-c opened the pull request:
<!-- Summarize your PR here. Keep it short and simple. -->
<!-- Explain existing problems or why this pull request is necessary -->
This PR updates PM5 to 1.21.80

## Backwards compatibility
<!-- Any possible backwards incompatible changes? How are they solved, or how can they be solved? -->
<!-- If not, you can delete this section -->
See https://github.com/pmmp/BedrockProtocol/pull/312

dries-c committed:
Bedrock 1.21.80 support

dktapps commented on src/world/biome/model/BiomeDefinitionEntryData.php:
```diff
@@ -0,0 +1,54 @@
+/**
+ * Model for loading biome definition entries data from JSON.
+ */
+final class BiomeDefinitionEntryData implements \JsonSerializable{
```
Don't think this is needed since all the fields are public anyway

dktapps commented on src/world/biome/model/BiomeDefinitionEntryData.php:
```diff
@@ -0,0 +1,54 @@
+	/**
+	 * @return mixed[]
+	 */
+	public function jsonSerialize() : array{
```
Same again here

dktapps commented on src/world/biome/model/ColorData.php:
```diff
@@ -0,0 +1,41 @@
+/**
+ * Model for loading color data from JSON.
+ */
+final class ColorData implements \JsonSerializable{
```
Same again here

dktapps commented on src/world/biome/model/ColorData.php:
```diff
@@ -0,0 +1,41 @@
+	/**
+	 * @return mixed[]
+	 */
+	public function jsonSerialize() : array{
```
and here

dktapps commented on src/world/biome/model/BiomeDefinitionEntryData.php:
```diff
@@ -0,0 +1,54 @@
+	 * @var string[]
+	 * @phpstan-var list<string>
+	 */
+	public array $tags;
```
These fields should all be marked as @required

dktapps commented on src/world/biome/model/ColorData.php:
```diff
@@ -0,0 +1,41 @@
+	public int $r;
+	public int $g;
+	public int $b;
+	public int $a;
```
These need to be marked as @required

dries-c committed:
Update bedrock-data

dries-c committed:
Add required tags to models

IcyEndymion004 commented:
Just tried running this, and got the following error

[Crash.txt](https://github.com/user-attachments/files/20093791/New.Text.Document.txt)

dktapps commented:
Yeah, that'll be because of this commit: https://github.com/pmmp/BedrockData/commit/a8932e40388c7280783ab78a8853d9491cb5dd40

dktapps committed:
Merge branch 'stable' into bedrock-1.21.80

dktapps committed:
Fixed biome data loading

dktapps committed:
Support newest world format
apparently I messed up the blockstate data version last time around... it hasn't changed since 1.21.60

dktapps committed:
always CS has to complain...

dktapps committed:
Sync with release versions

dktapps committed:
Ready 5.28.0 release

dktapps committed:
this might help...
